src/wwwroot/services/templates.js

```diff
@@ -69,14 +83,34 @@
         request.actionDescription = 'action_templates_creating',
         request.method = 'POST',
         request.url = RELAY_CONFIG.baseUrl + '/accounts/' + accountId + '/templates';
+
+        return $http(request).then(function (result) {
+          return result.createdResourceId;
+        });
       } else {
+        var body = request.data.body;
+        // Prevent to update body using deprecated property
+        request.data.body = "";
         request.actionDescription = '"Editing a template"',
         request.method = 'PUT',
         request.url = RELAY_CONFIG.baseUrl + '/accounts/' + accountId + '/templates/' + model.id;
+        return editTemplateBody(accountId, model.id, request.data.name, body).then(function (r) {
+          return $http(request).then(function (result) {
```

Comment:
Since you are refactoring this method, maybe it is a good idea refactoring it totally in place of keeping those weird things to reuse some code.

Reply:
👍